### PR TITLE
Log payment validation errors to Sentry

### PR DESCRIPTION
**Pull request by notbakaneko**
These used to be reported to a now non-existent channel, so this is about getting visibility on payment processing validation errors back (because standard `ValidationException`s don't get reported to sentry).

### CODE REVIEW

**Comment by notbakaneko** [comment]
I've been made aware of a whoops 👀

File: app/Libraries/Payments/PaymentProcessor.php [hunk 1]
```diff
@@ -341,11 +341,8 @@ protected function getOrder()
 
     /**
      * Convenience method that calls dispatchValidationFailed() and then throws the supplied exception.
-     *
-     * @param Exception $exception
-     * @return void
      */
-    protected function throwValidationFailed(Exception $exception)
+    protected function throwValidationFailed(Exception $exception): void
```
Comment (nanaya) [inline app/Libraries/Payments/PaymentProcessor.php:345]: this wrapper is kinda useless now with no actual channel? (along with all the event listeners/handlers)

File: app/Libraries/Payments/PaymentProcessor.php [hunk 2]
```diff
@@ -216,7 +216,7 @@ public function cancel()
         $this->sandboxAssertion();
 
         if (!$this->validateTransaction()) {
-            $this->throwValidationFailed(new PaymentProcessorException($this->validationErrors()));
+            $this->throwValidationFailed(new PaymentProcessorException($this->getOrder(), $this->validationErrors()));
```
Comment (nanaya) [inline app/Libraries/Payments/PaymentProcessor.php:219]: not related to this pr but it looks like there should be an `assertValidTransaction` which validates transaction and throw if not valid

**Review by nanaya — CHANGES_REQUESTED** [review]

File: app/Libraries/Payments/PaymentProcessor.php (hunk 1 above)
Comment (notbakaneko) [reply app/Libraries/Payments/PaymentProcessor.php:348]: Not doing it in this PR, there's going to be a separate cleanup for all this; this PR is only for getting the errors reported again.

File: app/Exceptions/InvalidSignatureException.php [hunk 3]
```diff
@@ -7,7 +7,21 @@
...
 
-class InvalidSignatureException extends Exception
+class InvalidSignatureException extends Exception implements HasExtraExceptionData
 {
+    public function __construct(string $message = '', private array $extras = [])
+    {
+        parent::__construct($message);
+    }
+
     // doesn't really contain anything
```
Comment (nanaya) [inline app/Exceptions/InvalidSignatureException.php:17]: not sure if needed

File: app/Exceptions/Store/FulfillmentException.php [hunk 4]
```diff
@@ -3,10 +3,10 @@
...
 // See the LICENCE file in the repository root for full licence text.
 
-namespace App\Libraries\Fulfillments;
+declare(strict_types=1);
 
-use App\Exceptions\ValidationException;
+namespace App\Exceptions\Store;
 
-class FulfillmentException extends ValidationException
+class FulfillmentException extends OrderException
```
Comment (nanaya) [inline app/Exceptions/Store/FulfillmentException.php:10]: was this caught in the controller before? 🤔 looks like the listeners are dispatched in sync and thus previously potentially caught by the controller?

**Review by nanaya — CHANGES_REQUESTED** [review]

File: app/Exceptions/Store/FulfillmentException.php (hunk 4 above)
Comment (notbakaneko) [reply app/Exceptions/Store/FulfillmentException.php:10]: That's a good point; originally it was planned to have the fulfillers eventually moved to workers so it wasn't intended to have their exceptions caught by the controller, but we never went through with that 🤔

Comment (nanaya) [reply app/Exceptions/Store/FulfillmentException.php:10]: I guess it should be handled correctly then? Either queue the event handlers or make such errors return the expected code (200?) so the payment processor receives the correct status.

**Review by nanaya — APPROVED** [review]
seems correct code wise but I can't quite manually test all the code paths, so 😬 🤞

**Pull request merged**